test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging.Testing;
+using Moq;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    public class AcceptedAtActionResultTests
+    {
+        public static TheoryData<object> ValuesData
+        {
+            get
+            {
+                return new TheoryData<object>
+                {
+                    null,
+                    "Test string",
+                    new object(),
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ValuesData))]
+        public void Constructor_InitializesStatusCodeAndValue(object value)
+        {
+            // Arrange 
+            var expectedUrl = "testAction";
+
+            // Act         
+            var result = new AcceptedAtActionResult(
+                actionName: expectedUrl,
+                controllerName: null,
+                routeValues: null, value: value);
+
+            // Assert
+            Assert.Equal(StatusCodes.Status202Accepted, result.StatusCode);
+            Assert.Same(value, result.Value);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValuesData))]
+        public async Task ExecuteResultAsync_SetsStatusCodeAndValue(object value)
+        {
+            // Arrange           
+            var expectedUrl = "testAction";
+            var httpContext = GetHttpContext();
+            var actionContext = GetActionContext(httpContext);
+            var urlHelper = GetMockUrlHelper(expectedUrl);
+
+            // Act
+            var result = new AcceptedAtActionResult(
+                actionName: expectedUrl,
+                controllerName: null,
+                routeValues: null,
+                value: value);
+
+            result.UrlHelper = urlHelper;
+            await result.ExecuteResultAsync(actionContext);
+
+            // Assert
+            Assert.Equal(StatusCodes.Status202Accepted, httpContext.Response.StatusCode);
+            Assert.Same(value, result.Value);
```

Comment:
If the 406 is a result of the formatter throwing, you could use Moq to make this work:

```C#
object actual = null;
var formatter = new Mock<IOutputFormatter>();
formatter.Setup(f => f.WriteAsync(It.IsAny<OutputFormatterWriteContext>()).Callback((OutputFormatterWriteContext context) =>
{
  actual = context.Object;
});

...
Assert.Same(expected, actual);
```